PlatformLevelTechempower/AsciiString.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Text;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.System;
+
+namespace PlatformLevelTechempower
+{
+    public struct AsciiString : IEquatable<AsciiString>
+    {
+        private readonly byte[] _data;
+
+        public AsciiString(byte[] bytes) => _data = bytes;
+
+        public AsciiString(string s) => _data = Encoding.UTF8.GetBytes(s);
```

Comment:
We have a faster routine than this. `HttpUtilities. GetAsciiStringNonNullCharacters`

Reply:
Haha, it using UTF8 when it shouldn't anyway. Changed to ASCII.

Changed conversion back to string to use `GetAsciiStringNonNullCharacters`